### PR TITLE
add nvim

### PR DESCRIPTION
**Pull request by jpx40**
Add nvim + possibilty to change color. Color changing is not complete yet, but we can use lazyvim for that, it has one function to change the whole theme. No need for advanced Plugins anymore. I will add an integration to the tools from this repo later. 

[colorsheme lazyvim](http://www.lazyvim.org/plugins/colorscheme)

### CODE REVIEW

**Comment by jpx40** [comment]
We need an backup for the config especially if ad neovim. This will prevent overwrites

**Comment by jpx40** [comment]
i haven't found nvim themes for all used themes of this repo what should i do with the rest.

**Comment by Enigma1309** [comment]
> i haven't found nvim themes for all used themes of this repo what should i do with the rest.

go with default or something famous like Catppuccin-mocha or Dracula or Nord

**Comment by jpx40** [comment]
> > i haven't found nvim themes for all used themes of this repo what should i do with the rest.
> 
> go with default or something famous like Catppuccin-mocha or Dracula or Nord

ok

**Comment by ffoebel** [comment]
Always great to have more people experience a properly configured (n)vim, but this should be skipped if neovim is already installed and especially configured.
Don’t mess with a devs config 😅

**Comment by jpx40** [comment]
> Always great to have more people experience a properly configured (n)vim, but this should be skipped if neovim is already installed and especially configured.
> Don’t mess with a devs config 😅

We need a backup to prevent this issue. I will start an discussion to discuss the status of backup. We might need an choice menu in the installer for neovim

File: Scripts/custom_hypr.lst
Comment (kRHYME7) [inline Scripts/custom_hypr.lst:1]: Consider moving the nvim deps on a separate file. e.g ``` nvim_deps.lst

**Comment by kRHYME7** [comment]
I have some suggestions regarding this change. I hope my claims are valid  (Please correct me ) 

+ Make this change atomic as possible (  Looks good ✔️)

+  Modular to the extent that users have an option to not overwrite their own config. I  happened to watch this video a while ago https://www.youtube.com/watch?v=LkHjJlSgKZY I think we can move our nvim configs somewhere e.g ``` ~/.configs/hyprvim ``` (whatever sounds best) then we can use alias to use the nvim  from ./hyprvim . Also I think it would be cool if we adapt the config switching. (So we can't messed with devs configs)

+ Considering having the color scheme per theme, we can modify this, right? http://www.lazyvim.org/plugins/colorscheme. 
Bonus, If this is a template, we can integrate wallbash for this.

+ The configurations are easy to manage, but the deps should be listed on and optional file ( too many package deps).   e.g. ``` nvim_dep.lst  ``` then we can just merge this into the final    ``` install_pkg.lst ```  only if users choose to populate the nvim. 

Okay that's my 2 cents.

**Comment by jpx40** [comment]
> I have some suggestions regarding this change. I hope my claims are valid (Please correct me )
> 
>     * Make this change atomic as possible (  Looks good ✔️)
> 
>     * Modular to the extent that users have an option to not overwrite their own config. I  happened to watch this video a while ago https://www.youtube.com/watch?v=LkHjJlSgKZY I think we can move our nvim configs somewhere e.g `~/.configs/hyprvim` (whatever sounds best) then we can use alias to use the nvim  from ./hyprvim . Also I think it would be cool if we adapt the config switching. (So we can't messed with devs configs)
> 
>     * Considering having the color scheme per theme, we can modify this, right? http://www.lazyvim.org/plugins/colorscheme.
>       Bonus, If this is a template, we can integrate wallbash for this.
> 
>     * The configurations are easy to manage, but the deps should be listed on and optional file ( too many package deps).   e.g. `nvim_dep.lst ` then we can just merge this into the final    `install_pkg.lst`  only if users choose to populate the nvim.
> 
> 
> Okay that's my 2 cents.

yeah, lazyvim gives access to color sheme through that function. Advantage over normal neovim is we don't have to configure everything manually. We have to preinstall all colorshemes, but only one is loaded at startup. The Configs of the .config/nvim dir are reloaded every start so all we have to do is changing the colorsheme file

**Comment by jpx40** [comment]
I will work again this weekend on the commit, had no time. Is it ok if I add some ruby code or lua code to automate stuff.

**Comment by prasanthrangan** [comment]
i think we need something like archinstall to record user choice for default apps before install.

**Comment by jpx40** [comment]
> i think we need something like archinstall to record user choice for default apps before install.

yeah, do you like ruby? i want do learn it. we could the programm in ruby

**Comment by prasanthrangan** [comment]
> yeah, do you like ruby? i want do learn it. we could the programm in ruby

I've no idea about ruby so i wont be able to help :(
I was thinking about python (like archinstall) as a wrapper with [simple-term-menu](https://github.com/IngoMeyer441/simple-term-menu)

**Comment by jpx40** [comment]
> > yeah, do you like ruby? i want do learn it. we could the programm in ruby
> 
> I've no idea about ruby so i wont be able to help :( I was thinking about python (like archinstall) as a wrapper with [simple-term-menu](https://github.com/IngoMeyer441/simple-term-menu)

Ruby is simble think python just slightly different syntax

**Comment by jpx40** [comment]
> > yeah, do you like ruby? i want do learn it. we could the programm in ruby
> 
> I've no idea about ruby so i wont be able to help :( I was thinking about python (like archinstall) as a wrapper with [simple-term-menu](https://github.com/IngoMeyer441/simple-term-menu)

`ruby
class Numeric
  def plus(x)
    self.+(x)
  end
end

y = 5.plus 6
# y is now equal to 11`

you just use end after an function

**Comment by ffoebel** [comment]
Python is almost guaranteed to be installed on any system, ruby would be a dependency you introduce.

**Comment by jpx40** [comment]
> Python is almost guaranteed to be installed on any system, ruby would be a dependency you introduce.

yeah, i just don't like pythons syntax

**Comment by jpx40** [comment]
> > yeah, do you like ruby? i want do learn it. we could the programm in ruby
> 
> I've no idea about ruby so i wont be able to help :( I was thinking about python (like archinstall) as a wrapper with [simple-term-menu](https://github.com/IngoMeyer441/simple-term-menu)

how should the cli look like? should we have an menu where the user could choose what he wants to install? we could make it integrate  in the existing tool.

**Comment by jpx40** [comment]
https://github.com/jpx40/hyperdots_installer    started doing smth

**Comment by prasanthrangan** [comment]
> https://github.com/jpx40/hyperdots_installer started doing smth

Great, thank you so much!

**Comment by jpx40** [comment]
> > https://github.com/jpx40/hyperdots_installer started doing smth
> 
> Great, thank you so much!

i chose rust if this ok. I can show you the logic of it if wanna. To use it you need to install Rust. https://www.rust-lang.org/learn/get-started

**Comment by jpx40** [comment]
> > https://github.com/jpx40/hyperdots_installer started doing smth
> 
> Great, thank you so much!

currently i write every thing to custom_hypr.lst so  we could start tho tool right before that step.

**Comment by jpx40** [comment]
@prasanthrangan can u add the theme switcher for me? Is there an way to disable it per application basis when not can we at it. Just simple booleans in themeswitch.

**Comment by prasanthrangan** [comment]
> @prasanthrangan can u add the theme switcher for me? Is there an way to disable it per application basis when not can we at it. Just simple booleans in themeswitch.

Sure happy to help... but what's the use of boolean switch per application? any specific reason??

Also, i just got an idea to improve theme switcher framework which can reuse swwwallbash to deploy theme based colors... I'm kinda stuck with work and exams this week, but let me dig into this next week!

**Comment by jpx40** [comment]
> > @prasanthrangan can u add the theme switcher for me? Is there an way to disable it per application basis when not can we at it. Just simple booleans in themeswitch.
> 
> Sure happy to help... but what's the use of boolean switch per application? any specific reason??
> 
> Also, i've got an idea to improve theme switcher framework which can reuse swwwallbash to deploy theme based colors... I'm kinda stuck with work and exams this week, but let me dig into this next week!

i just like my neovim in Dark mode so i wanna have the option to disabable stuff optionally

**Comment by prasanthrangan** [comment]
> i just like my neovim in Dark mode so i wanna have the option to disabable stuff optionally

hmm oki... the new framework should support it without introducing any new variables or checks!

**Comment by jpx40** [comment]
> > i just like my neovim in Dark mode so i wanna have the option to disabable stuff optionally
> 
> hmm oki... the new framework should support it without introducing any new variables or checks!
> I've created a dev branch named `hyde`... i'll try to implement the new themeswitcher next week :)

I already added some code regarding this   code is in the pull. I am mobile will sent rught commit later. Another improvement recommendation move code out of .config/hyperdots to .local/share/hyprdots. Just keep the most important configs in .config.

**Comment by jpx40** [comment]
> > i just like my neovim in Dark mode so i wanna have the option to disabable stuff optionally
> 
> hmm oki... the new framework should support it without introducing any new variables or checks! I've created a dev branch named `hyde`... i'll try to implement the new themeswitcher next week :)

i wanna try out some new stuff, we might profit from an projekt like toml bombardil or chezmoi. Dotfile manager.


https://oknozor.github.io/toml-bombadil/
https://www.chezmoi.io/

**Comment by mahaveergurjar** [comment]
Please resolve the branch conflicts so this PR can merge in this repo and also change the pr acc to hyde

**Comment by jpx40** [comment]
> Please resolve the branch conflicts so this PR can merge in this repo and also change the pr acc to hyde

will do at the weekend when i find time but changed can be hold back.

**Comment by rubiin** [comment]
Hyde-cli provides neovim support I guess

**Comment by mahaveergurjar** [comment]
Yes but only install nvim not config file because the nvim package name in custom app list that why hyde-cli show nvim option but only install nvim not nvim config  because the nvim config are not this repo 

240425_11h21m41s_screenshot

**Comment by mahaveergurjar** [comment]
@jpx40 It is possible to nvim theme disable and support kitty terminal theme

**Comment by mahaveergurjar** [comment]
OK, Thanks :)

**Comment by boulder-roller** [comment]
@T-Crypt another job for me? lol

**Pull request closed without merging**